src/actor.js

```diff
@@ -402,12 +361,22 @@ export const call = async (actId, input, options = {}) => {
     if (waitSecs <= 0) return run; // In this case there is nothing more to do.
 
     // Wait for run to finish.
-    const updatedRun = await waitForRunToFinish({
-        actId,
-        runId: run.id,
-        token,
-        waitSecs,
-    });
+    let updatedRun;
+    try {
+        updatedRun = await waitForRunToFinish({
+            actorId: actId,
+            runId: run.id,
+            token,
+            waitSecs,
+        });
+    } catch (err) {
+        throw new ApifyCallError({ id: run.id, actId }, 'Apify.call() failed, cannot fetch actor run details from the server');
```

Comment:
The exception can be something else, e.g. that there is no network connection, and here we'd swallow all error details. We should only swallow it if it's the `Waiting for run to finish failed` error, in other cases just rethrow the original exception.

Reply:
Good catch